src/RenderWebGL.js

```diff
@@ -685,6 +686,79 @@ class RenderWebGL extends EventEmitter {
     }
 
     /**
+     * @typedef ColorExtraction
+     * @property {Uint8Array} data Raw pixel data for the drawable
+     * @property {int} width Drawable bounding box width
+     * @property {int} height Drawable bounding box height
+     * @property {object} color Color object with RGBA properties at picked location
+     */
+
+    /**
+     * Return drawable pixel data and color at a given position
+     * @param {int} x The client x coordinate of the picking location.
+     * @param {int} y The client y coordinate of the picking location.
+     * @param {int} radius The client radius to extract pixels with.
+     * @return {?ColorExtraction} Data about the picked color
+     */
+    extractColor (x, y, radius) {
+        // @todo this doesn't seem to work for retina screens
```

Comment:
Yep - `_nativeSize` is always in terms of pixels (technically texels) whereas `canvas.client{Width,Height}` are in terms of HTML client "pixels" -- which aren't real pixels in HiDPI mode.

It might be a good idea to have this function take "Scratch coordinates" as input and do the conversion from client coordinates at the GUI layer, like this: https://github.com/LLK/scratch-gui/pull/606/files#diff-136323462d986e04bdd73357a9d350c3R71

Don't forget to scale the radius, too... if you're worried about non-square scaling, I think I'd switch this to take a width & height rather than two "radii" but I'm OK with one radius if you're not worried about non-square scaling.